File: packages/jsii-java-runtime/project/src/main/java/org/jsii/JsiiException.java
package org.jsii;

/**
 * An error raised by the jsii runtime.
 */
public final class JsiiException extends RuntimeException {

    /**
     * Creates an exception.
     * @param message The error message
     */
    public JsiiException(final String message) {
        super(message);
    }

    /**
     * Creates an exception.
     * @param e The error that caused this exception
     */
    public JsiiException(final Throwable e) {
        super(e);
    }

    /**
     * Creates an exception.
     * @param message The error message
     * @param e The error that caused this exception
     */
    public JsiiException(final String message, final Throwable e) {
        super(message, e);
    }
}

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/JsiiException.cs
using System;
using Amazon.JSII.JsonModel.Api.Response;

namespace Amazon.JSII.Runtime
{
    public abstract class JsiiException : Exception
    {
        public ErrorResponse? ErrorResponse { get; }
        public Type? ExceptionType { get; }

        internal JsiiException() : base()
        {
        }
        
        internal JsiiException(string message) : base(message)
        {
        }

        internal JsiiException(string message, Exception? innerException)
            : base(message, innerException)
        {
        }

        internal JsiiException(ErrorResponse response, Exception? innerException)
            : base(response.Error, innerException)
        {
            ErrorResponse = response;
        }
    }

    public sealed class JsiiError : JsiiException
    {
      internal JsiiError() : base()
      {
      }

      internal JsiiError(string message) : base(message)
      {
      }

      internal JsiiError(string message, Exception? innerException)
          : base(message, innerException)
      {
      }

      internal JsiiError(ErrorResponse response, Exception? innerException)
          : base(response, innerException)
      {
      }

    }
}
